src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJModelFactory.java

```diff
@@ -79,4 +88,48 @@ public Library createLibrary(@NotNull String name) {
     throw new UnsupportedOperationException(
         "Only common libraries like Scala SDK are currently supported.");
   }
+
+  /**
+   * This buddy here does all the nasty heavy lifting in order to check whether there are {@link
+   * Module} that depend on the removed {@link com.intellij.openapi.module.Module} and if finds,
+   * marks their {@link Module}s {@link StateMonitor} state.
+   *
+   * @param course            a {@link Course} which {@link Module}s to check
+   * @param removedModuleName a {@link String} name of the removed
+   * {@link com.intellij.openapi.module.Module}
+   */
+  public void markDependentModulesInvalid(@NotNull IntelliJCourse course,
+      @NotNull String removedModuleName) {
+    APlusProject project = course.getProject();
+    com.intellij.openapi.module.Module[] projectModules = project.getModuleManager().getModules();
+
+    for (com.intellij.openapi.module.Module module : projectModules) {
+      Module courseModule = (Module) course.getComponentIfExists(module.getName());
+      if (courseModule != null) {
+        List<String> courseModuleDependencies = getCourseModuleDependencies(courseModule);
+        if (courseModuleDependencies != null && !courseModuleDependencies.isEmpty()
+            && courseModuleDependencies.contains(removedModuleName)) {
+          courseModule.stateMonitor.set(StateMonitor.ERROR);
+        }
+      }
+    }
+  }
+
+  /**
+   * A simple wrapper method on {@link Module#getDependencies()} to hide the thrown {@link
+   * ComponentLoadException}.
+   *
+   * @param courseModule a {@link Module} to get dependencies from
+   * @return a {@link List} of {@link String}s representing names of the dependencies.
+   */
+  @Nullable
+  public List<String> getCourseModuleDependencies(@NotNull Module courseModule) {
+    List<String> courseModuleDependencies = null;
+    try {
+      courseModuleDependencies = courseModule.getDependencies();
+    } catch (ComponentLoadException ex) {
+      logger.error(ex.getMessage(), ex);
+    }
+    return courseModuleDependencies;
+  }
 }
```

Comment:
This is fine :+1: though I am thinking that `Module#getDependecies` should actually be changed so that it is called only after the module is loaded and instead of parsing IML file, it would use IDEA platform API to access dependency info. In that way, we could get rid of the exception altogether.